rdflib/graph.py

```diff
@@ -389,6 +389,27 @@ def add(self, triple):
         assert isinstance(o, Node), "Object %s must be an rdflib term" % (o,)
         self.__store.add((s, p, o), self, quoted=False)
 
+    def add_all(self , triples, all_or_none=True):
+        """Add triples with self as context
+
+        Parameters:
+        all_or_none -- if True, add triples only if all triples satisfy the assertion
+        else, add triples till the first unsatisfied assertion is encountered 
+        """
+
+        if(all_or_none):
+            for i, (s, p, o) in enumerate(triples,1):
+                assert isinstance(s, Node), "Subject %s in Triple %d must be an rdflib term" % (s,i,)
+                assert isinstance(p, Node), "Predicate %s in Triple %d must be an rdflib term" % (p,i,)
+                assert isinstance(o, Node), "Object %s in Triple %d must be an rdflib term" % (o,i,)
+
+        for i, (s, p, o) in enumerate(triples,1):
+            if(all_or_none==False):
```

Comment:
I wonder if there might be a way to optimise this when `all_or_none` is True, to avoid this `if` statement on every iteration of this loop (answer is always going to be the same every iteration).

Reply:
Yes I agree. I'll think of a better solution.

Reply:
@ashleysommer I have optimized the code and I feel it has better readability too. Is it fine now?

Reply:
Yes. Much better now. Thanks.